### PR TITLE
Build a universal binary to support M1 (Apple silicon)

### LINKED ISSUE #177: Apple Silicon run-time error
After successfully installing a version of rocksdb with #171 patched in ([buu700/rocksdb-tmp](https://github.com/buu700/rocksdb-tmp)), I ran into the following error on macOS 11.5.1:

```
[ 3:10PM] buu700@Cyph:~$ node
Welcome to Node.js v16.1.0.
Type ".help" for more information.
> require('rocksdb')
dyld: lazy symbol binding failed: Symbol not found: __Z20crc32c_runtime_checkv
  Referenced from: /Users/buu700/node_modules/rocksdb/build/Release/leveldown.node
  Expected in: flat namespace

dyld: Symbol not found: __Z20crc32c_runtime_checkv
  Referenced from: /Users/buu700/node_modules/rocksdb/build/Release/leveldown.node
  Expected in: flat namespace

Abort trap: 6
```

Edit: Looks like https://github.com/facebook/rocksdb/issues/7710 was the issue, in which case simply updating RocksDB should resolve this.

### PR DESCRIPTION
**Pull request by Nazgolze**
Closes #177 

Since this change is around cpu architecture and not OS, I did some testing on Ubuntu ARM64. Interestingly, 5.1.0 here did not have the same issue as on the Mac M1. Nevertheless, I've tested this PR on the M1 Mac and Ubuntu ARM64, have not encountered any errors and, so far, it's working well in our app (though the more eyeballs, the better - if @buu700 is able to give this a try as well :pray: )

### CODE REVIEW

**Comment by vweevers** [comment]
Can we make this work for a universal binary as well? See https://github.com/Level/leveldown/pull/781 for comparison - I want to do the same for rocksdb

File: deps/rocksdb/rocksdb.gyp [hunk 1]
```diff
@@ -173,6 +173,14 @@
                 , 'MACOSX_DEPLOYMENT_TARGET': '10.8'
             }
         }]
+      , ['target_arch == "arm64" or target_arch == "aarch64"', {
+          'sources': [
+            'rocksdb/util/crc32c_arm64.cc'
```
Comment (vweevers) [inline deps/rocksdb/rocksdb.gyp:178]: Can this `crc32c_arm64.cc` file be included unconditionally? By moving it to the main [sources](https://github.com/Level/rocksdb/blob/a54ad1e83782b600ffdfe3fa1bb142ae3e06a402/deps/rocksdb/rocksdb.gyp#L185).

The file has a [`#if defined(HAVE_ARM64_CRC)`](https://github.com/facebook/rocksdb/blob/09c7e96eac1ab983f97ce9e0406730b8014b3398/util/crc32c_arm64.cc#L8) so should already be a noop on other architectures.

**Review by vweevers — CHANGES_REQUESTED** [review]

File: deps/rocksdb/rocksdb.gyp (hunk 1 above)
Comment (Nazgolze) [reply deps/rocksdb/rocksdb.gyp:183]: I saw this just after I pushed just now :sweat_smile:
I'll have this in a sec :+1:

Comment (Nazgolze) [reply deps/rocksdb/rocksdb.gyp]: Pushed :+1:

File: binding.gyp [hunk 2]
```diff
@@ -38,14 +38,20 @@
                   , '-Wno-ignored-qualifiers'
                 ]
                 , 'OTHER_CPLUSPLUSFLAGS': [
-                    '-mmacosx-version-min=10.8'
+                    '-mmacosx-version-min=10.12'
```
Comment (vweevers) [inline binding.gyp:41]: Why 10.12?

Comment (Nazgolze) [reply binding.gyp:41]: I can change that back if desired :sweat_smile:
it was to match what was in `deps/rocksdb/rocksdb/build_tools/build_detect_platform`
but this was when I was debugging / testing

Comment (vweevers) [reply binding.gyp:41]: Let's keep it as-is here, to not make a breaking change.

**Review by vweevers — APPROVED** [review]
LGTM, thank you! Will release early next week

**Comment by vweevers** [comment]
Note to self: on release, compare prebuild binary size to the previous release

**Comment by vweevers** [comment]
5.1.1

**Pull request merged**